src/Raven.Client/Documents/Operations/Operation.cs

```diff
@@ -55,6 +55,8 @@ internal Operation(RequestExecutor requestExecutor, Func<IDatabaseChanges> chang
 
         private async Task Initialize()
         {
+            _result = _result.Task.IsCompleted ? new TaskCompletionSource<IOperationResult>(TaskCreationOptions.RunContinuationsAsynchronously) : _result;
```

Comment:
There is a race condition between executing 'WaitForCompletionAsync' and 'Initialize'. The 'Initialize' is executed in the background and it can swap the '_result'.

I would suggest creating a separate synchronous method e.g. 'InitializeResult' that will just do this and execute it in the 'WaitForCompletionAsync'.

Check also the behavior of 'OnError'

Reply:
done